### ISSUE TITLE
Decoder doesn't do anything when fed NAL units manually

### ISSUE THREAD

**Issue by ryangriggs** [mbebenita/Broadway#190]
I am attempting to feed in the NAL units from a h.264 file encoded with ffmpeg (baseline profile).  I have used node JS to split the NAL units based on the 00 00 00 01 header, into individual elements of an array.  Then I feed these to player.decode() one at a time.  The onPictureDecoded() callback never fires and the player canvas never changes from black.  I have tried two different h.264 files, same result.  No error messages or console logs for either one.  I have tested using data including the 00 00 00 01 NAL header, and also excluding this header.

I would be very appreciative of any advice to understand why it's not decoding the data, or what I'm doing wrong.

Here's the code (in the <body>):

```
// A manually-generated list of NAL units in original order, base64 encoded:
var nals = ["AAAAAA<snip>", "AAAAA<snip>", <snip about 10MB of NAL units>];

// Set up a player:
var player = new Player({
		useWorker: false,
		webgl: false
	});
document.body.append(player.canvas);
// At this point, a black canvas appears in the browser window.

// Set up onPictureDecoded callback:
player.decoder.onPictureDecoded = () => {
    console.log('onPictureDecoded!');
};

// Dump the NALs into the decode function
nals.forEach((data) => {
    player.decode(atob(data));  // Convert from base64 back to binary, then pass to decode()
});

// Nothing happens...
```

**Comment by ryangriggs** [comment]
Found the issue: I had to convert the NAL units to Uint8Array before passing to the decoder.  Works now.  Thx!

**Comment by njw-bupt** [comment]
@ryangriggs Hello. Could you please show me what's the NAL units that you put into the function decode() look like? I input the Uint8Array [0, 0, 1, 98, 65, 154, 224, ...] to Player.decode, but nothing happens.

**Comment by soliton4** [comment]
i think broadway expects the nal seperator 0, 0, 0, 1
it usually works when you fill in the missing 0
unless your nals are slices which will probably not work

**Comment by soliton4** [comment]
yeah slices should be fine as long as you keep them all in one bigger nal
its usually 0,0,0,1 for the main nal and then 0,0,1 as seperators within the nal for the different slices.
just keep the slices together and it will work in broadway

**Comment by njw-bupt** [comment]
Hello @soliton4! I have another question. Could you please give me an advice? 
I noticed that this project does not support weighted prediction for P-frames. If I have a H264 bit stream that contains I-frame and P-frame, it means I cannot use the Broadway to decode the video?

**Comment by soliton4** [comment]
I frames and p frames are supported
